straxen/plugins/acqmon_processing.py

```diff
@@ -240,6 +242,160 @@ def compute(self, aqmon_hits, start, end):
         result = result[sort]
         return result
 
+@export
+class GPS_sync(strax.Plugin):
+    """
+    Correct the event times to GPS time. 
+      1. Finds the TTL GPS pulses coming into the AM from the gps 
+    module and their pairs in the ASCII file coming from the module
+    for the correspondant run.
+      2. Corrects the timestamp of all events by linearly interpolating
+    between the previous and next sync pulses. 
+    """
+        
+    __version__ ='0.1.0'
+    depends_on = ('aqmon_hits', 'event_info')
+    provides  = 'gps_sync'
+    data_kind = 'gps_sync'
+    
+    def infer_dtype(self):
+        dtype = [(('Time since unix epoch [ns]', 'time'), np.int64),
+                 (('GPS absolute time [ns]', 't_gps'), np.int64),
+                ]
+        return dtype
+
+
+    def gps_times_from_runid(self,run_id):
+        '''
+        Fetches the mongodb looking for the gps_sync collection for 
+        timestamps between the start and end times of a given run_id.
+        '''
+
+        rundb = utilix.xent_collection()
+        gps_times = utilix.xent_collection(collection='gps_sync',  database='xenonnt')
+
+        if isinstance(run_id, str):
+            run_id = int(run_id)
+
+        query = {"number": run_id}
+
+        doc = rundb.find_one(query, projection={'start': 1, 'end': 1})
+        
+        start_t = doc['start'].replace(tzinfo=datetime.timezone.utc).timestamp()
+        end_t = doc['end'].replace(tzinfo=datetime.timezone.utc).timestamp()
+        
+        query = {"gps_sec": { '$gte': start_t - 1, '$lte': end_t + 1}, 'channel': 0 }
+
+        return pd.DataFrame(gps_times.find(query))
+
+    def cut_outside_run(self,aqmon_array, gps_array):
+        first_idx = None
+        last_idx = None
+        _idx = 0
+        while (first_idx == None) or (last_idx == None):
+
+            if np.abs(aqmon_array[0]-gps_array[_idx])<5e9:
+                first_idx = _idx
+            if np.abs(aqmon_array[-1]-gps_array[-_idx-1])<5e9:
+                last_idx = -_idx
+            _idx+=1
+        if last_idx == 0:
+            gps_array_corr = gps_array[first_idx:]
+        else:
+            gps_array_corr = gps_array[first_idx:last_idx]
+
+        return gps_array_corr
+
+    # Basic algorithm to compute corrected time given all the inputs
+    def compute_time_individual(self, t0_daq, t1_daq, t0_gps, t1_gps, t_evt):
+        '''
+        Compute the corrected time given the previous and next sync pulse.
+          * t0: previous pulse (daq or gps)
+          * t1: previous pulse (daq or gps)
+          * t_evt: uncorrected time of event (in DAQ time)
+        Returns a single value, t_evt in gps-time.
+        '''
+        delta_sync_offset = t0_gps-t0_daq
+        delta_t_evt = t_evt - t0_daq
+        
+        return t0_daq + delta_sync_offset + (t1_gps-t0_gps)/(t1_daq-t0_daq)*delta_t_evt
+
+    def compute_time_array(self, l_daq_sync, l_gps_sync, l_daq_evt):
```

Comment:
Did you measure the average time this function takes? is it worth using numba or is it most of the time spent loading the data from the db anyway?